app-samples/Cognitive-Services-Face-WPF/Sample-WPF/Controls/UIHelper.cs

```diff
@@ -124,19 +124,51 @@ public static IEnumerable<Face> CalculateFaceRectangleForRendering(IList<Detecte
                 uiWidth = (int)(ratio * uiHeight);
             }
 
-            int uiXOffset = (maxSize - uiWidth) / 2;
-            int uiYOffset = (maxSize - uiHeight) / 2;
-            float scale = (float)uiWidth / imageWidth;
+            var uiXOffset = (maxSize - uiWidth) / 2;
+            var uiYOffset = (maxSize - uiHeight) / 2;
+            var scale = (float)uiWidth / imageWidth;
 
             foreach (var face in faces)
             {
+                var left = (int)(face.FaceRectangle.Left * scale + uiXOffset);
+                var top = (int)(face.FaceRectangle.Top * scale + uiYOffset);
+
+                // Angle of face rectangles, default value is 0 (not rotated).
+                double faceAngle = 0;
+
+                // If head pose attributes have been obtained, re-calculate the left & top (X & Y) positions.
+                if (face.FaceAttributes?.HeadPose != null)
+                {
+                    // Head pose's roll value acts directly as the face angle.
+                    faceAngle = face.FaceAttributes.HeadPose.Roll;
+                    var angleToPi = Math.Abs((faceAngle / 180) * Math.PI);
+
+                    // _____       | / \ |
+                    // |____|  =>  |/   /|
+                    //             | \ / |
+                    // re-calculate the face rectangle left/top(X/Y) position.
```

Comment:
`Re-calculate the face rectangle's left & top (X & Y) positions.`

Reply:
Done.